RFC/src/RFC-0310_AssetTemplates.md

```diff
@@ -0,0 +1,218 @@
+| **Signatures**                     |               |                                                                                                                               |
+| creator_sig                        | `u256`        | A digital signature of the preceding 2 sections’ data using the asset creator’s private key corresponding to the `issuer` PKH |
+| **template-specific data**         |               | Depending on the template, the data in the final section varies and corresponds to the initial asset state                    |
+
+#### Committee parameters
+
+If `committee_mode` is `PERMISSIONED` the `committee_mode` object is
+
+| Name             | Type         | Description |
+|:-----------------|:-------------|:------------|
+| trusted_node_set | Array of PKH | See below   |
+
+Only the nodes in the trusted node set will be allowed to execute instructions for this asset.
+
+If `committee_mode` is `PERMISSIONLESS` the `committee_mode` object is
+
+| Name                    | Type  | Description                                                                                                                       |
+|:------------------------|:------|:----------------------------------------------------------------------------------------------------------------------------------|
+| node_threshold          | `u32` | The total number of validator nodes that can register to execute instructions for this asset                                      |
+| minimum_collateral      | `u64` | The minimum amount of Tari a validator node must put up in collateral in order to execute instructions for this asset.([RFC-???]) |
+| node_selection_strategy | `u32` | The selection strategy to employ allowing nodes to register to manage this asset                                                  |
+
+
+#### Issuer
+
+Anyone can issue assets on the Tari network from their wallet or client program. The wallet will provide the PKH and
+sign the instruction. The wallet needn’t use the same private key each time.
+
+#### Name and Description
+
+These fields are purely for informational purposes and do not need to be unique, and do not act as an asset ID.
+
+#### Top level digital issuers
+
+If it is likely that a digital asset issuer will be issuing many assets on the Tari Network (hundreds, or thousands),
+the issuer should strongly consider registering a name (e.g. `TARILABS`) as a top-level digital asset issuer. This is
+done via a special transaction on the base layer, described in [RFC-???](). A registered TLDI prevents spoofing of
+assets from copy cats or other malicious actors. It also makes asset discovery simpler.
+
+TLDI owners need to provide a valid signature proving that they own the given domain when creating assets.
+
+#### Asset identification
+
+Assets are identified by `<TLDI>.#_hhhhh` on the Tari network, where `#` is the template number, and `hhhhh`
```

Comment:
64bit is much long than 10 bits decimal. u64 = 0 to 18,446,744,073,709,551,615

Reply:
Yeah, we don't need fixed lengths, the delimiters `#` and `_` allow easy parsing of the ID

Reply:
But this does make me think that we should perhaps move the feature flags to the bottom of the reserved range. As it stands all beta template will have HUGE decimal numbers...

Reply:
Is that perhaps not a good thing? Having beta stuff sit way out there at the top of the range, while leaving the bottom lower numbers for normal use?

Reply:
That's exactly what I'm suggesting, but instead of

`100000000000000000000000000010101101101111` = 219,354,354,454,334,763,324
 `^- Beta flag `

we have
`000000000000001000000000000010101101101111` = 817,123,343,102
`~~~~~~~~~~~~~~^- Beta flag `

and reserve the higher bits for future feature flags that may/may not ever be used.

_numbers are made up and are for illustrative purposes only_

Reply:
This makes sense for now, but if we use the future flags in the future, this will not?

Reply:
Future flags will make the number bigger, but later. 

Maybe the asset id should split the template id up. Something like

`<TLDI>_<template_type>.<version>.<flags>_<hash>`

Reply:
`TLDI` was changed to `RAID_ID`, so my original comment is not applicable anymore.